scripts/upgrade

```diff
@@ -16,128 +16,150 @@ ynh_script_progression --message="Loading installation settings..." --weight=1
 
 app=$YNH_APP_INSTANCE_NAME
 
-domain=$(ynh_app_setting_get --app="$app" --key=domain)
-path_url=$(ynh_app_setting_get --app="$app" --key=path)
-admin=$(ynh_app_setting_get --app="$app" --key=admin)
+domain=$(ynh_app_setting_get --app=$app --key=domain)
+path_url=$(ynh_app_setting_get --app=$app --key=path)
+admin=$(ynh_app_setting_get --app=$app --key=admin)
 is_public=$(ynh_app_setting_get --app="$app" --key=is_public)
-final_path=$(ynh_app_setting_get --app="$app" --key=final_path)
-language=$(ynh_app_setting_get --app="$app" --key=language)
-wiki_name=$(ynh_app_setting_get --app="$app" --key=wiki_name)
-db_name=$(ynh_app_setting_get --app="$app" --key=db_name)
-db_pwd=$(ynh_app_setting_get --app="$app" --key=mysqlpwd)
+final_path=$(ynh_app_setting_get --app=$app --key=final_path)
+language=$(ynh_app_setting_get --app=$app --key=language)
+wiki_name=$(ynh_app_setting_get --app=$app --key=wiki_name)
+db_name=$(ynh_app_setting_get --app=$app --key=db_name)
+db_pwd=$(ynh_app_setting_get --app=$app --key=mysqlpwd)
 
 # Note(decentral1se): avoid using this on upgrade for the versions
 # of the application that upgrade and have not stored this in their
 # settings (when it was not available to them). Later on, when we have
 # moved a few versions on, we can re-enable this
-# wiki_name=$(ynh_app_setting_get --app="$app" --key=wiki_name)
+# wiki_name=$(ynh_app_setting_get --app=$app --key=wiki_name)
 
+#=================================================
+# CHECK VERSION
+#=================================================
+ynh_script_progression --message="Checking version..."
+
+upgrade_type=$(ynh_check_app_version_changed)
+
+#=================================================
+# BACKUP BEFORE UPGRADE THEN ACTIVE TRAP
+#=================================================
+ynh_script_progression --message="Backing up the app before upgrading (may take a while)..." --weight=6
+
+# Backup the current version of the app
+ynh_backup_before_upgrade
+ynh_clean_setup () {
+	# Restore it if the upgrade fails
+	ynh_restore_upgradebackup
+}
+# Exit if an error occurs during the execution of the script
+ynh_abort_if_errors
+
+#=================================================
+# STANDARD UPGRADE STEPS
 #=================================================
 # ENSURE DOWNWARD COMPATIBILITY
 #=================================================
 ynh_script_progression --message="Ensuring downward compatibility..." --weight=1
 
 # Fix is_public as a boolean value
 if [ "$is_public" = "Yes" ]; then
-    ynh_app_setting_set --app="$app" --key=is_public --value=1
+    ynh_app_setting_set --app=$app --key=is_public --value=1
     is_public=1
 elif [ "$is_public" = "No" ]; then
-    ynh_app_setting_set --app="$app" --key=is_public --value=0
+    ynh_app_setting_set --app=$app --key=is_public --value=0
     is_public=0
```

Comment:
So I guess this could be removed ? Is it not in the example_ynh anymore.

Reply:
Yes It should be remove if we want to use new system permission